include/envoy/network/listen_socket.h

```diff
@@ -82,6 +82,7 @@ class Socket {
      */
     struct Details {
       SocketOptionName name_;
+      absl::optional<std::string> description_;
```

Comment:
The details class we had in mind didn't really stand up well to the original_src socket option. Since it doesn't actually set a linux option on the socket, there was nothing to describe with it. So, I added a "description" field which we can use to textually describe it for UT. I'm not sure this is ideal. If anyone else has suggestions, I'm will to take them.

Reply:
Yeah seems this is only used for unit tests and potentially logging? I'm not sure if we really want this.

Reply:
I might be able to just apply all the options and see if the source address is updated. It's not as nice as having it tell us what it is, since that's really all I care about here, but it's probably better than adding something out of character.